pkg/ottl/contexts/internal/ottlcommon/value.go

```diff
@@ -51,40 +51,37 @@ func SetValue(value pcommon.Value, val interface{}) {
 	case []byte:
 		value.SetEmptyBytes().FromRaw(v)
 	case []string:
-		value.Slice().RemoveIf(func(_ pcommon.Value) bool {
-			return true
-		})
+		value.SetEmptySlice()
 		for _, str := range v {
 			value.Slice().AppendEmpty().SetStr(str)
 		}
```

Comment:
Add a `value.Slice().EnsureCapacity(len(v))` same to all cases to reduce number of allocations.

Reply:
Good catch. Done.